src/client.rs

```diff
@@ -50,14 +51,60 @@ impl Client {
         //     });
         // }
         // Ok(Channel::balance_list(endpoints.into_iter()))
-        let end_points = cfg.endpoints.join(",");
+        if cfg.endpoints.is_empty() {
+            panic!("Empty etcd endpoints");
+        }
+        let mut end_points = cfg.endpoints.join(",");
         let env = Arc::new(EnvBuilder::new().build());
+        if cfg.endpoints.len() > 1 {
+            let socket_address: SocketAddr = cfg
+                .endpoints
+                .first()
+                .unwrap_or_else(|| panic!("Fail to get the first endpoint"))
+                .parse()
+                .unwrap_or_else(|e| {
+                    panic!(
+                        "Fail to parse enpoint to socket address, the error is {}",
+                        e,
+                    )
+                });
+            cfg.endpoints.iter().for_each(|x| {
+                let is_same_schema = Self::is_same_ip_address_schema(
+                    socket_address,
+                    x.parse().unwrap_or_else(|e| {
+                        panic!(
+                            "Fail to parse enpoint to socket address, the error is {}",
+                            e,
+                        )
+                    }),
+                );
+                if !is_same_schema {
+                    panic!("Fail to get the first endpoint")
+                }
+            });
+
+            if socket_address.is_ipv4() {
```

Comment:
check address type before for each loop

Reply:
The for each loop is used to check the endpoints has same type of ip address schema, if it is failed, then it just panic. So we should not check address type before for each loop